aten/src/ATen/native/cuda/RowwiseScaledMM.cu

```diff
@@ -402,10 +405,11 @@ void f8f8bf16_rowwise_impl_sm100(
           cutlass::gemm::collective::StageCountAutoCarveout<static_cast<int>(sizeof(typename CollectiveEpilogue::SharedStorage))>,
           MainloopScheduleType>::CollectiveOp;
 
-  using GemmKernel = cutlass::gemm::kernel::GemmUniversal<
-      cute::Shape<int, int, int>,
-      CollectiveMainloop,
-      CollectiveEpilogue>;
+  using GemmKernel = at::cuda::detail::enable_3x_kernel_for_sm10_or_later<
```

Comment:
Noob q, so this just wrapping makes it so that when not on the correct arch the call operator is a noop? how does that end up helping

Reply:
Compilation of this file gets faster (I'm seeing 30-40% speedups, for the case when building for single arch), which is good as this file was taking loads of time (several minutes on my machine) to compile.  On the other hand, we cannot hit the noop as at run-time, depending on the actual GPU, the execution is routed to proper kernel.

(With the [first commit](https://github.com/pytorch/pytorch/pull/149978/commits/c4ab9842c12f5eb25eaa6bc776989b7daedcad0a), I tried with instantiating template per SM arch in different files, that would cut the compile time of this functionality to more than half - if done in parallel.  However, this approach seems cumbersome to me, and I think for now we're good this way.)

Reply:
If compile time is still a problem here, we could proceed this way to split template instantiations into multiple files, one file per arch.